src/security/credentials.rs

```diff
@@ -65,6 +68,22 @@ pub enum CertificateRequestType {
         GRPC_SSL_REQUEST_AND_REQUIRE_CLIENT_CERTIFICATE_AND_VERIFY as u32,
 }
 
+/// User-provided callback function for reload cert. [`None`] indicates that
+/// no reloading is needed, and [`Some(ServerCredentialsBuilder)`] indicates
+/// that reloading is needed.
+pub trait UserFetcher {
+    fn fetch(&mut self)
+        -> std::result::Result<Option<ServerCredentialsBuilder>, Box<dyn StdError>>;
+}
+
+#[repr(C)]
+pub struct CertUserData {
+    initial_cert_cfg: Option<Box<grpc_ssl_server_certificate_config>>,
```

Comment:
It's unnecessary to store initial certificates. `UserFetcher` should be able to handle it properly.

Reply:
The initial cert is convenient for the user to operate correctly. If it is completely handed over to the user, user may forget to set the initial(means using previous cert) and cannot load any certificates in the first time.
Anyway, I have removed it. Tell me if your opinion changes